From a115f1a0443b4e0b4dcd1cb037095eabb37d86c6 Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:17 +0100
Subject: [PATCH 10/51] Revert "Store token literal data in PCH. Avoids disk
 read on PreProc expansion (ROOT-6794)."

This reverts commit 6033555158d81c37a1375c468de082823781c440.
---
 .../clang/include/clang/Serialization/ASTReader.h     |  3 ---
 .../clang/lib/Serialization/ASTReader.cpp             | 11 -----------
 .../clang/lib/Serialization/ASTWriter.cpp             |  3 ---
 3 files changed, 17 deletions(-)

diff --git a/interpreter/llvm-project/clang/include/clang/Serialization/ASTReader.h b/interpreter/llvm-project/clang/include/clang/Serialization/ASTReader.h
index 1362e736b1..62c25f5b7a 100644
--- a/interpreter/llvm-project/clang/include/clang/Serialization/ASTReader.h
+++ b/interpreter/llvm-project/clang/include/clang/Serialization/ASTReader.h
@@ -676,9 +676,6 @@ private:
   /// files.
   llvm::DenseSet<LoadedMacroInfo> LoadedUndefs;
 
-  /// \Token literal data loaded and owned by us.
-  std::vector<std::string *> TokenLiteralDataLoaded;
-
   using GlobalMacroMapType =
       ContinuousRangeMap<serialization::MacroID, ModuleFile *, 4>;
 
diff --git a/interpreter/llvm-project/clang/lib/Serialization/ASTReader.cpp b/interpreter/llvm-project/clang/lib/Serialization/ASTReader.cpp
index 64fc3ea3af..a994219164 100644
--- a/interpreter/llvm-project/clang/lib/Serialization/ASTReader.cpp
+++ b/interpreter/llvm-project/clang/lib/Serialization/ASTReader.cpp
@@ -1830,14 +1830,6 @@ Token ASTReader::ReadToken(ModuleFile &M, const RecordDataImpl &Record,
     if (IdentifierInfo *II = getLocalIdentifier(M, Record[Idx++]))
       Tok.setIdentifierInfo(II);
   }
-
-  if (Tok.isLiteral()) {
-    const RecordData& RD = reinterpret_cast<const RecordData&>(Record);
-    std::string* Lit = new std::string(ReadString(RD, Idx));
-    TokenLiteralDataLoaded.push_back(Lit);
-    Tok.setLiteralData(Lit->c_str());
-  }
-
   return Tok;
 }
 
@@ -10269,9 +10261,6 @@ ASTReader::ASTReader(Preprocessor &PP, InMemoryModuleCache &ModuleCache,
 ASTReader::~ASTReader() {
   if (OwnsDeserializationListener)
     delete DeserializationListener;
-  for (auto PStr: TokenLiteralDataLoaded) {
-     delete PStr;
-  }
 }
 
 IdentifierResolver &ASTReader::getIdResolver() {
diff --git a/interpreter/llvm-project/clang/lib/Serialization/ASTWriter.cpp b/interpreter/llvm-project/clang/lib/Serialization/ASTWriter.cpp
index ba44c40208..8ec82ebbc1 100644
--- a/interpreter/llvm-project/clang/lib/Serialization/ASTWriter.cpp
+++ b/interpreter/llvm-project/clang/lib/Serialization/ASTWriter.cpp
@@ -4519,9 +4519,6 @@ void ASTWriter::AddToken(const Token &Tok, RecordDataImpl &Record) {
     // is needed.
     AddIdentifierRef(Tok.getIdentifierInfo(), Record);
   }
-
-  if (Tok.isLiteral())
-    AddString(StringRef(Tok.getLiteralData(), Tok.getLength()), Record);
 }
 
 void ASTWriter::AddString(StringRef Str, RecordDataImpl &Record) {
-- 
2.43.5

